src/net.cpp

```diff
@@ -831,6 +831,33 @@ void SocketSendData(CNode* pnode)
     pnode->vSendMsg.erase(pnode->vSendMsg.begin(), it);
 }
 
+void CheckOffsetDisconnectedPeers(const CNetAddr& ip)
+{
+    static std::set<CNetAddr> setOffsetDisconnectedPeers;
+    int nConnections = 0;
+    {
+        LOCK(cs_vNodes);
+        nConnections = (int) vNodes.size();
```

Comment:
`vNodes` can have "bad" peers that will be disconnected as soon as they send the version message. Them are pushed back into the vector as soon as the socket is connected, independently on if them are good or not.

 So, if I'm understanding well this flow, the `nConnections` is looking for the number of good peers connected and should use another "established connections" subset or filter the `vNodes` set to only get the valid connected peers number.